vault/resource_quota_lease_count_test.go

```diff
@@ -56,6 +56,45 @@ func TestQuotaLeaseCount(t *testing.T) {
 	})
 }
 
+func TestQuotaLeaseCountWithRole(t *testing.T) {
+	name := acctest.RandomWithPrefix("lease-count")
+	backend := acctest.RandomWithPrefix("approle")
+	role := acctest.RandomWithPrefix("test-role")
+	leaseCount := "1001"
+	newLeaseCount := "2001"
+	resourceName := "vault_quota_lease_count.foobar"
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testutil.TestEntPreCheck(t) },
```

Comment:
Along with the comments above, I think we also need this addition. At the end of TF tests, the SDK will attempt to compare what you have provided in the config with what is returned from Vault (the import test). Since we are always providing the `role` as part of the config, the SDK will try to look for it even on Vault 1.11 versions. Therefore, this test by design should fail on versions < 1.12. To fix this, we can let the CI know that this is a version specific test using the following:

```suggestion
		PreCheck:  func() { 
		         testutil.TestEntPreCheck(t)
			SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion112)
		 },
```

Reply:
Added here https://github.com/hashicorp/terraform-provider-vault/pull/1994/commits/9ff1f50adf9dae1d239d973a5fdcfc2bf1ada3a6